### PR TITLE
Simplified getKeysFileUrls, reduced # of calls to the framework

### PR DESCRIPTION
**Pull request by smcmurtry**
- removed generator, 
- made a simpler `getPeriodsSinceLastFetch` fn to replace `keysSinceLastFetch`
- reduced # of calls to the framework by calling `this.backendInterface.retrieveDiagnosisKeys` with `0` when the app is 1st installed, instead of 14 times

### CODE REVIEW

File: src/services/ExposureNotificationService/ExposureNotificationService.ts [hunk 1]
```diff
@@ -411,20 +396,16 @@ export class ExposureNotificationService {
     return this.finalize();
   }
 
-  private async getKeysFileUrls() {
+  private async getKeysFileUrls(): Promise<any> {
     const currentStatus = this.exposureStatus.get();
```
Comment (jeberhardt) [inline src/services/ExposureNotificationService/ExposureNotificationService.ts:400]: Can `currentStatus` be passed into this function? A tiny bit more functional programming that way.

File: src/services/ExposureNotificationService/ExposureNotificationService.ts [hunk 2]
```diff
@@ -411,20 +396,16 @@ export class ExposureNotificationService {
...
     let lastCheckedPeriod = currentStatus.lastChecked?.period;
-    while (true) {
-      const {value, done} = await generator.next();
-      if (done) break;
-      if (!value) continue;
-      const {keysFileUrl, period} = value;
-      captureMessage('loop period', {period});
+    const periodsSinceLastFetch = this.getPeriodsSinceLastFetch(lastCheckedPeriod);
+    for (const period of periodsSinceLastFetch) {
+      const keysFileUrl = await this.backendInterface.retrieveDiagnosisKeys(period);
```
Comment (maxneuvians) [inline src/services/ExposureNotificationService/ExposureNotificationService.ts:405]: It looks like the old code wrapped this in a try catch - the danger here might be that a server or network error goes unnoticed

Comment (smcmurtry) [reply src/services/ExposureNotificationService/ExposureNotificationService.ts]: Good catch!

**Review by maxneuvians — APPROVED** [review]
Nice work!

**Pull request merged**